akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -73,8 +80,12 @@ private[akka] class Association(
     if (queue eq null)
       queue = Source.queue(256, OverflowStrategy.dropBuffer)
         .to(transport.outbound(this)).run()(materializer)
-    if (systemMessageQueue eq null)
-      systemMessageQueue = Source.queue(256, OverflowStrategy.dropBuffer)
-        .to(transport.outboundSystemMessage(this)).run()(materializer)
+    if (systemMessageQueue eq null) {
+      val matValue = Source.queue(256, OverflowStrategy.dropBuffer)
+        .toMat(transport.outboundSystemMessage(this))(Keep.both)
+        .run()(materializer)
+      systemMessageQueue = matValue._1
```

Comment:
maybe destructure matValue first with pattern matching? This is not a hot path, the allocation does not matter.